src/compiler/crystal/macros.cr

```diff
@@ -2303,15 +2303,33 @@ module Crystal::Macros
     end
   end
 
-  # A macro expression,
-  # surrounded by {{ ... }} (output = true)
-  # or by {% ... %} (output = false)
-  # class MacroExpression < ASTNode
-  # end
+  # A macro expression.
+  #
+  # Every expression *node* is equivalent to:
+  #
+  # ```
+  # {% if node.output? %}
+  #   \{{ {{ node.exp }} }}
+  # {% else %}
+  #   \{% {{ node.exp }} %}
+  # {% end %}
+  # ```
+  class MacroExpression < ASTNode
+    # Returns the expression inside this node.
+    def exp : ASTNode
```

Comment:
We might want to rename this as well.

```suggestion
    def expression : ASTNode
```

Reply:
There are `UnaryExpression#exp`, `VisibilityModifier#exp`, `ControlExpression#exp`, `MacroFor#exp`, and `AsmOperand#exp`, but no uses of `#expression` at all. So for consistency this is going to stay

Reply:
Oh well, I guess we're stuck with it then. Maybe we could do some general cleanup later.